src/super_gradients/training/transforms/processing.py

```diff
@@ -0,0 +1,193 @@
+        postprocessed_predictions = predictions
+        for processing, metadata in zip(self.processings[::-1], metadata.metadata_lst[::-1]):
+            postprocessed_predictions = processing.postprocess_predictions(postprocessed_predictions, metadata)
+        return postprocessed_predictions
+
+
+class ImagePermute(Processing):
+    """Permute the image dimensions.
+
+    :param permutation: Specify new order of dims. Default value (2, 0, 1) suitable for converting from HWC to CHW format.
+    """
+
+    def __init__(self, permutation: Tuple[int, int, int] = (2, 0, 1)):
+        self.permutation = permutation
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, None]:
+        processed_image = np.ascontiguousarray(image.transpose(*self.permutation))
+        return processed_image, None
+
+    def postprocess_predictions(self, predictions: np.ndarray, metadata: None) -> np.ndarray:
+        return predictions
+
+
+class NormalizeImage(Processing):
+    """Normalize an image based on means and standard deviation.
+
+    :param mean:    Mean values for each channel.
+    :param std:     Standard deviation values for each channel.
+    """
+
+    def __init__(self, mean: List[float], std: List[float]):
+        self.mean = np.array(mean).reshape((1, 1, -1)).astype(np.float32)
+        self.std = np.array(std).reshape((1, 1, -1)).astype(np.float32)
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, None]:
+        return (image - self.mean) / self.std, None
+
+    def postprocess_predictions(self, predictions: np.ndarray, metadata: None) -> np.ndarray:
+        return predictions
+
+
+class DetectionCenterPadding(Processing):
+    """Preprocessing transform to pad image and bboxes to `output_shape` shape (H, W).
+    Center padding, so that input image with bboxes located in the center of the produced image.
+
+    Note: This transformation assume that dimensions of input image is equal or less than `output_shape`.
+
+    :param output_shape: Output image shape (H, W)
+    :param pad_value:   Padding value for image
+    """
+
+    def __init__(self, output_shape: Tuple[int, int], pad_value: int):
+        self.output_shape = output_shape
+        self.pad_value = pad_value
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, DetectionPadToSizeMetadata]:
+        pad_top, pad_bot, pad_left, pad_right = _get_center_padding_params(input_shape=image.shape, output_shape=self.output_shape)
+        processed_image = _pad_image(image, (pad_top, pad_bot), (pad_left, pad_right), self.pad_value)
+
+        return processed_image, DetectionPadToSizeMetadata(pad_top=pad_top, pad_left=pad_left)
+
+    def postprocess_predictions(self, predictions: np.ndarray, metadata: DetectionPadToSizeMetadata) -> np.ndarray:
+        return _shift_bboxes(targets=predictions, shift_h=-metadata.pad_top, shift_w=-metadata.pad_left)
+
+
+class DetectionSidePadding(Processing):
```

Comment:
How is it different from DetectionCenterPadding? I assume it pad image from the right & bottom?
If so, there should be a docstring reflecting this.
Also perhaps we can simplify these two classes into this to avoid duplication?:
```
@dataclass
class PaddingParams:
    pad_top: int
    pad_left: int
    pad_bottom: int
    pad_right: int


class DetectionPadding(Processing):
    def __init__(self, output_shape: Tuple[int, int], pad_value: int):
        self.output_shape = output_shape
        self.pad_value = pad_value

    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, DetectionPadToSizeMetadata]:
        padding_params = _get_center_padding_params(input_shape=image.shape, output_shape=self.output_shape)
        processed_image = _pad_image(image, padding_params, self.pad_value)

        return processed_image, DetectionPadToSizeMetadata(padding_params)

    def postprocess_predictions(self, predictions: np.ndarray, metadata: DetectionPadToSizeMetadata) -> np.ndarray:
        return _shift_bboxes(targets=predictions, padding_params)
    
    @abc.abstractmethod
    def _get_padding_params(self, input_shape) -> PaddingParams:
        pass


class DetectionCenterPadding(DetectionPadding):
    def __init__(self, output_shape: Tuple[int, int], pad_value: int):
        self.output_shape = output_shape
        self.pad_value = pad_value

    def _get_padding_params(self, input_shape):
        return _get_center_padding_params(input_shape=image.shape, output_shape=self.output_shape)


class DetectionBottomRightPadding(DetectionPadding):
    def __init__(self, output_shape: Tuple[int, int], pad_value: int):
        self.output_shape = output_shape
        self.pad_value = pad_value

    def _get_padding_params(self, input_shape):
        return _get_bottom_right_padding_params(input_shape=image.shape, output_shape=self.output_shape)

```

Reply:
DOne something similar

https://github.com/Deci-AI/super-gradients/blob/3229c5447e8af8ffc44550fb7245670c8c2ef430/src/super_gradients/training/transforms/processing.py#L152-L204